test/Datadog.Trace.TestHelpers/HeadersCollectionTestExtensions.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Net;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.TestHelpers
+{
+    public abstract class HeadersCollectionTestExtensions
```

Comment:
why not as a regular static class? e.g. `public static class HeadersCollectionTestHelper`
My rule of thumb: use inheritance if polymorphism is needed. Otherwise, using "composition" or helper functions is more maintainable.

Reply:
It feeds XUnit MemberData. It's also possible to use ClassData, but then you need 3 classes.

Reply:
Agree, didn't focus on the that poor name choice. I think it wanted to be extension originally 😄 Renamed.